### DISCUSSION TITLE
nix: wayland-protocols override causes rebuild of qt

### DISCUSSION THREAD

**Discussion by Cloudef** [hyprwm/Hyprland#5945] [General]
This commit currently causes local rebuild of Qt even though I'm not using the hyprland overlay but instead hyprland package directly. Is hyprland depending on Qt somewhere? Even if I don't pull xdg-desktop-portal-hyprland (which uses Qt) the rebuild is still being triggered, even going as far to rebuilding  python3.11-pyqt5 :thinking:.
https://github.com/hyprwm/Hyprland/commit/a4f38a07d7578f0b06c09de7d04682e0aaddb12a

**Comment by fufexan** [comment]
Are you sure xdph flake version isn't pulled in *at* all in your configuration? You can use `nix-tree` to check that.

**Comment by Cloudef** [comment]
Thanks for the tip but it seems `nix-tree` first requires a build of the flake before it can operate on it.

```nix
    nixpkgs.overlays = [(final: prev: {
      hyprland = inputs.hyprland.packages.${pkgs.system}.hyprland;
      hyprland-debug = inputs.hyprland.packages.${pkgs.system}.hyprland-debug;
      # xdg-desktop-portal-hyprland = inputs.hyprland.packages.${pkgs.system}.xdg-desktop-portal-hyprland;
    })];

    programs.hyprland.enable = true;
    programs.hyprland = {
      package = cfg.finalPackage;
      portalPackage = pkgs.xdg-desktop-portal-hyprland;
    };
```
This is how I'm using hyprland, the xdg-desktop-portal-hyprland override in the overlay is commented out. I do notice the nixpkgs programs.hyprland module will override xdg-desktop-portal-hyprland and set the correct hyprland input for it, but xdph's default.nix doesn't seem to inherit wayland-protocols from it, so it should still be fine.

**Reply by fufexan** [reply]
I'm not sure what's causing it to be honest.
Apart from https://github.com/Cloudef/nixos-flake/blob/master/flake.nix#L15 I don't think there's anything that may be causing a rebuild. But even if you remove that, you don't seem to be using the Cachix, so Nix will still build all the deps locally.

**Reply by Cloudef** [reply]
I already tried to remove that too, but no luck :/ Also I do use cachix
```
    nix.settings.substituters = [ "https://hyprland.cachix.org" ];
    nix.settings.trusted-public-keys = [ "hyprland.cachix.org-1:a7pgxzMz7+chwVL3/pzj6jIBMioiJM7ypFP8PwtkuGc=" ];
```

**Comment by Cloudef** [comment]
wayland-protocols override was removed so this problem is no longer relevant. However the current master does not build on nix due to cmake expecting `subprojects/udis86` folder (even with ?submodules=1)

**Reply by fufexan** [reply]
What's the type of the url? `github:` or `git+https://`? Only the latter works.

**Reply by Cloudef** [reply]
Thanks `git+https://github.com/hyprwm/Hyprland?submodules=1` works in flake input
